apps/laboratory/src/pages/api/auth/[...nextauth].ts

```diff
@@ -47,22 +45,19 @@ export default async function auth(req: NextApiRequest, res: NextApiResponse) {
           if (!credentials?.message) {
             throw new Error('SiweMessage is undefined')
           }
-          const siwe = new SiweMessage(credentials.message)
-          const provider = new ethers.JsonRpcProvider(
-            `https://rpc.walletconnect.com/v1?chainId=eip155:${siwe.chainId}&projectId=${projectId}`
-          )
-          const nonce = await getCsrfToken({ req: { headers: req.headers } })
-          const result = await siwe.verify(
-            {
-              signature: credentials?.signature || '',
-              nonce
-            },
-            { provider }
-          )
+          const { message, signature } = credentials
+          const address = message.match(/0x[a-fA-F0-9]{40}/u)?.[0] || ''
+          const chainId = `eip155:${
+            credentials.message.match(/Chain ID: (?<temp1>\d+)/u)?.[1] || 1
```

Comment:
If we can get the message to match the SiweMessage spec you can parse the message.
```js
const parsedMessage = new SiweMessage(message)
const chainId = `eip155:${parsedMessage.chainId}`
```

Reply:
`siwe` doesn't support `ReCaps` thus failing validation